File: packages/typia/src/transformers/features/json/JsonCreateIsStringifyTransformer.ts
import { JsonIsStringifyProgrammer } from "@typia/core";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace JsonCreateIsStringifyTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.factory({
      ...props,
      method: "json.stringify",
      write: JsonIsStringifyProgrammer.write,
    });
}

File: packages/transform/src/features/json/JsonCreateIsStringifyTransformer.go
package json

import (
	corejson "github.com/samchon/typia/packages/core/src/programmers/json"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var JsonCreateIsStringifyTransformer = jsonCreateIsStringifyTransformerNamespace{}

type jsonCreateIsStringifyTransformerNamespace struct{}

func (jsonCreateIsStringifyTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return corejson.JsonStringifyProgrammer.Write(call.TypeText)
}
